gobench.go

```diff
@@ -94,509 +96,519 @@ func logStatsCSV(name string, times []time.Duration) {
+
+        var conn driver.Connection
+        var err error
+        if protocol == "HTTP" {
+                connConfig := http.ConnectionConfig{
+                        Endpoints: []string{endpoint},
+                        ConnLimit: nrConnections,
+                }
+                if usetls {
+                        connConfig.TLSConfig = &tls.Config{
+                                InsecureSkipVerify: true,
+                        }
+                }
+                conn, err = http.NewConnection(connConfig)
+                if err != nil {
+                        log.Fatalf("Failed to create HTTP connection: %v", err)
+                }
+        } else if protocol == "VST" {
+                connConfig := vst.ConnectionConfig{
+                        Endpoints: []string{endpoint},
+                        Transport: vstproto.TransportConfig{
+                                ConnLimit: nrConnections,
+                        },
+                }
+                if usetls {
+                        connConfig.TLSConfig = &tls.Config{
+                                InsecureSkipVerify: true,
+                        }
+                }
+                conn, err = vst.NewConnection(connConfig)
+                if err != nil {
+                        log.Fatalf("Failed to create HTTP connection: %v", err)
+                }
+        } else {
+                log.Fatalf("-protocol needs to be HTTP or VST")
+        }
+
+        clientConfig := driver.ClientConfig{
+                Connection: conn,
+        }
+
+        if username != "" {
+                clientConfig.Authentication = driver.BasicAuthentication(username, password)
+        }
+
+        c, err := driver.NewClient(clientConfig)
+
+        db, err := c.Database(nil, "benchDB")
+        if err != nil {
+                // Create a database
+                db, err = c.CreateDatabase(nil, "benchDB", nil)
+                if err != nil {
+                        log.Fatalf("Failed to create database: %v", err)
+                }
+        }
+
+        // Create collection
+        col, err := db.Collection(nil, "test")
+        if err != nil {
+                col, err = db.CreateCollection(nil, "test", nil)
+                if err != nil {
+                        log.Fatalf("Failed to create collection: %v", err)
+                }
+        }
+
+        startTime := time.Now()
+        switch testcase {
+        case "postDocs":
+                doPostDocs(col)
+        case "seedDocs":
+                doSeedDocs(col)
+        case "readDocs":
+                doReadDocs(col)
+        case "readSameDocs":
+                doReadSameDocs(col)
+        case "replaceDocs":
+                doReplaceDocs(col)
+        case "readThreeDiamondAQL":
+                AQLdb, AQLcol := doInitThreeDiamondAQL(c)
+                startTime = time.Now()
+                doReadThreeDiamondAQL(AQLdb, AQLcol)
+        case "all":
+                doPostDocs(col)
```

Comment:
Does startTime need to be resetted before each doXXX()?

Reply:
No, it does not, as the measurements have their own `startTime` (though named the same, its a different variable scope unless I horribly misunderstand golang's scoping).

Alas, for the `all` target the timing and reporting is now all mucked up; I should fix that.

Reply:
I changed the bookeeping. Not an ideal solution; for the future I should come up with a bit of a refactor.

Reply:
Now good!